.github/workflows/verify_consumer_pacts.yaml

```diff
@@ -0,0 +1,205 @@
+            echo "git checkout -b ${{ env.CHECKOUT_BRANCH }} ${{ env.CHECKOUT_SHA }}"
+            git checkout -b ${{ env.CHECKOUT_BRANCH }} ${{ env.CHECKOUT_SHA }} || echo "already in ${{ env.CHECKOUT_BRANCH }}"
+            echo "git branch"
+            git branch
+          else
+            if [[ "${{ github.event_name }}" == "push" ]] || [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+              echo "git checkout ${{ env.CURRENT_BRANCH }}"
+              git checkout ${{ env.CURRENT_BRANCH }}
+            else
+              echo "git checkout -b ${{ env.CURRENT_BRANCH }} ${{ env.CURRENT_SHA }}"
+              git checkout -b ${{ env.CURRENT_BRANCH }} ${{ env.CURRENT_SHA }}
+            fi
+          fi
+          echo "git rev-parse HEAD"
+          git rev-parse HEAD
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ github.ref }}-${{ github.sha }}
+          restore-keys: v1-${{ runner.os }}-gradle-${{ github.ref }}
+
+      - name: Verify consumer pacts and publish verification status to Pact Broker
+        id: verification-test
+        env:
+          PACT_PROVIDER_COMMIT: ${{ env.PROVIDER_SHA }}
+          PACT_PROVIDER_BRANCH: ${{ env.PROVIDER_BRANCH }}
+          PACT_BROKER_USERNAME: ${{ secrets.PACT_BROKER_USERNAME }}
+          PACT_BROKER_PASSWORD: ${{ secrets.PACT_BROKER_PASSWORD }}
+        run: |
+          echo "provider-sha=${{ env.PROVIDER_SHA }}" >> $GITHUB_OUTPUT
+          echo "env.CHECKOUT_BRANCH=${{ env.CHECKOUT_BRANCH }} # If not empty, this reflects the branch being checked out (generated by Pact Broker)"
+          echo "env.CHECKOUT_SHA=${{ env.CHECKOUT_SHA }}       # If not empty, this reflects the git commit hash of the branch being checked out (generated by Pact Broker)"
+          echo "env.CURRENT_BRANCH=${{ env.CURRENT_BRANCH }}   # This reflects the branch being checked out if CHECKOUT_BRANCH is empty"
+          echo "env.CURRENT_SHA=${{ env.CURRENT_SHA }}         # This reflects the git commit hash of the branch being checked out if CHECKOUT_BRANCH is empty"
+          echo "env.PROVIDER_BRANCH=${{ env.PROVIDER_BRANCH }} # This reflects the provider branch for pact verification"
+          echo "env.PROVIDER_SHA=${{ env.PROVIDER_SHA }}       # This reflects the provider version for pact verification"
+          echo "env.CONSUMER_BRANCH=${{ env.CONSUMER_BRANCH }} # This reflects the consumer branch for pact verification (generated by Pact Broker)"
+          echo "env.CONSUMER_SHA=${{ env.CONSUMER_SHA }}       # This reflects the consumer version for pact verification (generated by Pact Broker)"
+          ./gradlew --build-cache verifyPacts --scan
+
```

Comment:
You might want to add in a task to upload the test reports after verify is concluded.  I found the errors produced without the corresponding report details to be inscrutable in GitHub without it:
```
      - name: Upload Test Reports
        if: always()
        uses: actions/upload-artifact@v1
        with:
          name: Test Reports
          path: service/build/reports
```

Reference [PR](https://github.com/DataBiosphere/terra-workspace-manager/pull/1511/files#diff-beac71d47300076c72974aa63384728f206fd6f49fd016afcba5d51a7d57adecR190-R195)

After this runs with artifact uploads, the test reports with more detailed stacktrace will be available for download from the "Summary" of GHA results page
<img width="245" alt="image" src="https://github.com/DataBiosphere/jade-data-repo/assets/111856/81ad13e8-c41a-4682-b796-11f7307be1d0">

Reply:
Good idea!  But in this case, we are already doing something similar with Gradle scans:

https://scans.gradle.com/s/floparrttw5hu/tests/task/:verifyPacts/details/bio.terra.pact.provider.DataRepoProviderTest?top-execution=1